logstash-core/src/main/java/org/logstash/config/ir/PipelineIR.java

```diff
@@ -51,7 +51,7 @@ public PipelineIR(Graph inputSection, Graph filterSection, Graph outputSection,
 
         this.graph.validate();
 
-        if (this.getOriginalSource() != null && this.getOriginalSource().matches("^\\S+$")) {
+        if (this.getOriginalSource() != null && !this.getOriginalSource().matches("^\\s+$")) {
```

Comment:
Notice the negation, this tests that the source is not entirely composed of whitespace. If it is entirely composed of whitespace we do the more expensive graph computation.